packages/cashier/src/Stores/verification-store.js

```diff
@@ -0,0 +1,130 @@
+import { action, observable } from 'mobx';
+import Constants from 'Constants/constants';
+import ErrorStore from './error-store';
+
+export default class VerificationStore {
+    constructor({ WS, root_store }) {
+        this.root_store = root_store;
+        this.WS = WS;
+    }
+
+    @observable is_button_clicked = false;
+    @observable timeout_button = '';
+    @observable error = new ErrorStore();
+    @observable is_email_sent = false;
+    @observable is_resend_clicked = false;
+    @observable resend_timeout = 60;
+
+    @action.bound
+    setIsButtonClicked(value) {
+        this.is_button_clicked = value;
+    }
+
+    @action.bound
+    setTimeoutButton(value) {
+        this.timeout_button = value;
+    }
+
+    @action.bound
+    setIsEmailSent(value) {
+        this.is_email_sent = value;
+    }
+
+    @action.bound
+    setIsResendClicked(value) {
+        this.is_resend_clicked = value;
+    }
+
+    @action.bound
+    setResendTimeout(value) {
+        this.resend_timeout = value;
+    }
+
+    clearTimeoutVerification() {
+        if (this.timeout_button) {
+            clearTimeout(this.timeout_button);
+        }
+    }
+
+    @action.bound
+    setTimeoutVerification() {
+        this.clearTimeoutVerification();
+        this.setTimeoutButton(
+            setTimeout(() => {
+                this.clearVerification();
+            }, 3600000)
```

Comment:
That's a pretty long timeout, why?

Reply:
This timeout was there from the previous implementations and since my changes themselves were quite large, I tried not to change business methods as much as possible to prevent extra issues, so I prefer not to change it in this PR.